persistence/src/main/scala/akka/persistence/EventStoreInterop.scala

```diff
@@ -55,29 +53,29 @@ object EventStoreInterop {
           persistence.journalFor(journalPluginId)
         }
       }
-    } yield new EventStore[F, A] {
+    } yield new EventStore[F, Any] {
 
       val persistenceId = eventSourcedId.value
 
-      override def events(fromSeqNr: SeqNr): F[sstream.Stream[F, EventStore.Persisted[A]]] = {
+      override def events(fromSeqNr: SeqNr): F[sstream.Stream[F, EventStore.Persisted[Any]]] = {
 
         trait Consumer {
-          def onEvent(event: EventStore.Persisted[A]): F[Consumer]
+          def onEvent(event: EventStore.Persisted[Any]): F[Consumer]
         }
 
         sealed trait State
         object State {
 
-          object Empty                                                                 extends State
-          case class Buffering(events: Vector[EventStore.Event[A]])                    extends State
-          case class Consuming(consumer: F[Consumer])                                  extends State
-          case class Finishing(events: Vector[EventStore.Event[A]], finalSeqNr: SeqNr) extends State
+          object Empty                                                                   extends State
+          case class Buffering(events: Vector[EventStore.Event[Any]])                    extends State
+          case class Consuming(consumer: F[Consumer])                                    extends State
+          case class Finishing(events: Vector[EventStore.Event[Any]], finalSeqNr: SeqNr) extends State
 
         }
 
-        def event(persisted: PersistentRepr): F[EventStore.Event[A]] =
+        def event(persisted: PersistentRepr): F[EventStore.Event[Any]] =
           for {
-            payload <- MonadThrow[F].catchNonFatal(persisted.payload.asInstanceOf[A])
+            payload <- MonadThrow[F].catchNonFatal(persisted.payload)
```

Comment:
I do not think you need `catchNonFatal` here anymore, as it was only needed to catch `asInstanceOf` failing with a `ClassCastException` error.

Reply:
done, thanks